### ISSUE TITLE
[Feature Request] object-curly-newline { "multiline-only-require": true }

### ISSUE THREAD

**Issue by svenssonaxel** [eslint/eslint#14598]
**What rule do you want to change?**
object-curly-newline

**Does this change cause the rule to produce more or fewer warnings?**
`multiline-only-require` would produce fewer than `multiline`.

**How will the change be implemented? (New option, new default behavior, etc.)?**
A new object option: `"multiline-only-require": true` requires line breaks if there are line breaks inside properties or between properties. Otherwise, anything is allowed. If this option is combined with `"consistent": true`, then consistency is enforced no matter whether there are any line breaks inside properties or between properties.

**Please provide some example code that this change will affect:**

```js
const a = {
  b: c,
}
```

**What does the rule currently do for this code?**
With `"multiline": true`, it is changed to:
```js
const a = { b: c }
```

**What will the rule do after it's changed?**
Leave it as is.

**Are you willing to submit a pull request to implement this change?**
Not currently.
